### PR TITLE
5x constructor

### PR DESCRIPTION
**Pull request by markstory**
This shouldn't be an impactful breaking change as this class was marked `internal` and is exempt from backwards compat.

If folks are ok with this kind of change, I'll do some more as it does let us trim down the size of the library which could help with opcache performance.

### CODE REVIEW

File: src/Http/Client/FormDataPart.php [hunk 1]
```diff
@@ -93,12 +65,12 @@ class FormDataPart implements Stringable
...
      * @param string|null $charset The charset of the data.
      */
-    public function __construct(string $name, string $value, string $disposition = 'form-data', ?string $charset = null)
-    {
-        $this->_name = $name;
-        $this->_value = $value;
-        $this->_disposition = $disposition;
-        $this->_charset = $charset;
+    public function __construct(
+        protected string $name,
```
Comment (ADmad) [inline src/Http/Client/FormDataPart.php:69]: Can we have inline docblocks for the props here if needed and do the popular tools / IDE read them?

File: tests/bootstrap.php [hunk 2]
```diff
@@ -145,3 +147,8 @@
     $loader = new SchemaLoader();
     $loader->loadInternalFile(env('FIXTURE_SCHEMA_METADATA'));
 }
+
+FactoryLocator::add(
+    'Table',
+    (new TableLocator())->allowFallbackClass(false)
```
Comment (ADmad) [inline tests/bootstrap.php]: Why? 🙂

File: src/Http/Client/FormDataPart.php (hunk 1 above)
Comment (othercorey) [reply src/Http/Client/FormDataPart.php:69]: Inline docblocks are allowed, but I don't know if it's any different than having the @param tags.

**Comment by othercorey** [comment]
The mixed promoted and non-promoted properties are what felt awkward to me. I suppose we'll just get used to it over time.

However, what I found very hard to read was constructors with some parameters promoted and some not.

File: src/Http/Client/FormDataPart.php (hunk 1 above)
Comment (ADmad) [reply src/Http/Client/FormDataPart.php:69]: That's good enough I guess. Should only be needed when we want to specify array shapes/types for e.g.

File: tests/bootstrap.php (hunk 2 above)
Comment (markstory) [reply tests/bootstrap.php]: I wasn't able to run `vendor/bin/phpunit tests/TestCase/Http` without this.

**Comment by markstory** [comment]
> However, what I found very hard to read was constructors with some parameters promoted and some not.

Agreed, I'd like to avoid that too.

File: tests/bootstrap.php (hunk 2 above)
Comment (ADmad) [reply tests/bootstrap.php]: Strange, it's causing various other failures since we don't have concrete classes for all the tables used in tests.

Comment (markstory) [reply tests/bootstrap.php]: Derp, the allowFallbackClass() part shouldn't be there.

File: src/Http/Client/FormDataPart.php [hunk 3]
```diff
@@ -253,8 +225,8 @@ protected function _headerParameterToString(string $name, string $value): string
     {
         $transliterated = Text::transliterate(str_replace('"', '', $value));
         $return = sprintf('%s="%s"', $name, $transliterated);
-        if ($this->_charset !== null && $value !== $transliterated) {
-            $return .= sprintf("; %s*=%s''%s", $name, strtolower($this->_charset), rawurlencode($value));
+        if ($this->charset !== null && $value !== $transliterated) {
+            $return .= sprintf("; %s*=%s''%s", $name, strtolower($this->charset), rawurlencode($value));
```
Comment (othercorey) [inline src/Http/Client/FormDataPart.php:229]: Seems like this code should be tested unless codecov is wrong.

**Comment by markstory** [comment]
I'll take the lack of feedback as a :+1: and I'll continue to look for more of these.

**Pull request merged**